### PR TITLE
organizations role

### LINKED ISSUE #1109: organizations role
##### SUMMARY
Role for creating/managing organizations via variables.
(eg. https://github.com/redhat-cop/tower_configuration/tree/devel/roles/organizations)

##### ISSUE TYPE
 - Feature Idea

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: roles/organizations/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+================================
+
+This role creates and manages Organizations.
+
+Role Variables
+--------------
+
+This role supports the [Common Role Variables](https://github.com/theforeman/foreman-ansible-modules/blob/develop/README.md#common-role-variables).
+
+The main data structure for this role is the list of `foreman_organizations`. Each `activation_key` requires the following fields:
```
Comment (ehelms) [inline roles/organizations/README.md]:
```suggestion
The main data structure for this role is the list of `foreman_organizations`. Each `organization` requires the following fields:
```

File: roles/organizations/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+theforeman.foreman.organizations
+================================
+
+This role creates and manages Organizations.
```
Comment (ehelms) [inline roles/organizations/README.md]:
```suggestion
This role creates and manages organizations.
```

Or capitalize it elsewhere

File: roles/organizations/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+- `name`: The name of the organization.
+
+The following fields are optional in the sense that the server will use default values when they are omitted:
+
+- `label`: The label of the organization.
+- `description`: The description of the organization.
+- `state`: The state of the organization. Can be `present` or `absent`.
+
+### Organization Data Structure
+|`state`|`present`|no|Desired State of the Entity.|
```
Comment (ehelms) [inline roles/organizations/README.md]:
```suggestion
```

File: roles/organizations/README.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+```yaml
+--- 
+- name: add organizations to foreman
+  hosts: localhost
+  gather_facts: false
+  tasks:
+    - name: add organizations
+      include_role: 
+        name: theforeman.foreman.organizations
+      vars:
```
Comment (ehelms) [inline roles/organizations/README.md:31]:
```suggestion
  roles:
    - name: add organizations
      theforeman.foreman.organizations:
```

File: roles/organizations/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+- name: Add organizations
+  theforeman.foreman.organization:
+    name:                               "{{ foreman_organizations_item.name }}"
+    description:                        "{{ foreman_organizations_item.description | default(omit) }}"
+    label:                              "{{ foreman_organizations_item.label | default(omit) }}"
+    server_url:                         "{{ foreman_server_url | default(omit) }}"
+    username:                           "{{ foreman_username | default(omit) }}"
+    password:                           "{{ foreman_password | default(omit) }}"
+    validate_certs:                     "{{ foreman_validate_certs | default(omit) }}"
+    state:                              "{{ foreman_organizations_item.state | default(omit) }}"
```
Comment (ehelms) [inline roles/organizations/tasks/main.yml]: Weird spacing here

File: changelogs/fragments/1109-organizations-role.yml [hunk 6]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - new ``organizations`` role to manage organizations (https://github.com/theforeman/foreman-ansible-modules/issues/1109)
```
Comment (ehelms) [inline changelogs/fragments/1109-organizations-role.yml:2]: Is double `` intentional?

Comment (evgeni) [reply changelogs/fragments/1109-organizations-role.yml:2]: yes, it ends up in reStructuredText which uses double for code…

File: roles/organizations/README.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+            foreman_password: changeme
+            foreman_organizations: 
+              - name: raleigh
+                label: rdu
+                state: present
+              - name: default
+                label: boring
+                state: absent
+              - name: lanai 
+                description: pacific datacenter
```
Comment (ehelms) [inline roles/organizations/README.md]: I think this is indented one level deeper than it should be.

**Review by ehelms — APPROVED** [review]
Fix the indentation and good once tests are passing for me

File: roles/organizations/README.md (hunk 7 above)
Comment (evgeni) [reply roles/organizations/README.md]: actually two

**Pull request merged**